packages/migrate/src/utils/ensureDatabaseExists.ts

```diff
@@ -8,78 +8,132 @@ import {
   uriToCredentials,
 } from '@prisma/internals'
 import chalk from 'chalk'
-import type execa from 'execa'
-import prompt from 'prompts'
 
 export type MigrateAction = 'create' | 'apply' | 'unapply' | 'dev' | 'push'
-export type DbType = 'MySQL' | 'PostgreSQL' | 'SQLite' | 'SQL Server' | 'CockroachDB'
+export type PrettyProvider = 'MySQL' | 'PostgreSQL' | 'SQLite' | 'SQL Server' | 'CockroachDB' | 'MongoDB'
 
 // TODO: extract functions in their own files?
 
-export async function getDbInfo(schemaPath?: string): Promise<{
-  schemaWord: 'database' // legacy? could be removed?
+export type DatasourceInfo = {
   name?: string // from datasource name
+  prettyProvider?: PrettyProvider // pretty name for the provider
   url?: string // from getConfig
   dbLocation?: string // host without credentials
-  dbType?: DbType // pretty name
   dbName?: string // database name
-  schema?: string // only for postgres right now (but SQL Server has this concept too)
-}> {
-  const datamodel = await getSchema(schemaPath)
-  const config = await getConfig({ datamodel, ignoreEnvVarErrors: false })
-  const activeDatasource = config.datasources?.[0]
+  schema?: string // database schema (!= multiSchema, can be found in the connection string like `?schema=myschema`)
+  schemas?: string[] // database schemas from the datasource (multiSchema preview feature)
+}
+
+export async function getDatasourceInfo({
+  schemaPath,
+  throwIfEnvError,
+}: { schemaPath?: string; throwIfEnvError?: boolean } = {}): Promise<DatasourceInfo> {
+  const schema = await getSchema(schemaPath)
+  let config: ConfigMetaFormat
+
+  // Try parsing the env var if defined
+  // Because we want to get the database name from the url later in the function
+  // If it fails we try again but ignore the env var error
```

Comment:
In which case does this make sense to fail here?

Reply:
When we want to keep the current behavior where commands error when the env var is undefined with: 
```
 Prisma schema validation - (get-config wasm)
 Error code: P1012
 error: Error validating datasource \`db\`: the URL must start with the protocol \`postgresql://\` or \`postgres://\`.
   -->  schema.prisma:5
    | 
  4 |   provider = "postgresql"
  5 |   url      = env("SOME_DEFINED_INVALID_URL")
    | 
 
 Validation Error Count: 1
 [Context: getConfig]
```